matminer/featurizers/base.py

```diff
@@ -359,9 +359,21 @@ def __init__(self, featurizers):
     def featurize(self, *x):
         return np.hstack(np.squeeze(f.featurize(*x)) for f in self.featurizers)
 
+    def set_n_jobs(self, n_jobs):
+        for f in self.featurizers:
+            f.set_n_jobs(n_jobs)
+
     def feature_labels(self):
         return sum([f.feature_labels() for f in self.featurizers], [])
 
+    def featurize_dataframe(self, df, col_id, ignore_errors=False,
+                            return_errors=False, inplace=True):
+        for f in self.featurizers:
+            df = f.featurize_dataframe(df, col_id, ignore_errors,
+                                       return_errors, inplace)
+            df[f.feature_labels()] = df[f.feature_labels()].applymap(np.squeeze)
+        return df
```

Comment:
This change is going to wreck havoc with caching. If I understand multiprocessing correctly, each time featurize_dataframe completes the processes holding the cache will exit - thereby preventing data cached from one featurizer to be employed by another.

So, I think we need to be really sure we want this change to featurize data frame. Can you convince me? 

Fwiw, I'm not concerned FunctionFeaturize need work with MultiFeaturizer. I picture it being used in a Pipleline after primary features are computed but not in parallel (as assumed by MultiFeaturizer)

Reply:
Hmm... this seems like an important problem but I don't know how I'd fix it off of the top of my head.  Any featurizer that overrides featurize_dataframe will end up needing something like this, though (I don't know if any of the other featurizers do this).

To your last point, my main reason for trying to make this work is more practical, which is to say that someone actually tried to use it that way, it didn't work, and they e-mailed us to ask why. I think new users encountering a lack of interoperability of the code is bad for its perceived robustness, but I also think you're right that the more important supported use case should be in sequential applications of featurization, rather than parallel.

Reply:
Also, can you give me an example where this causes the problem you mention in a significant way?  I'm confident there is one, I just don't know the ins and outs of featurizers that use caching well enough to come up with an example on my own.

Reply:
The Voronoi notebook in our [examples repo](https://github.com/hackingmaterials/matminer_examples/blob/master/notebooks/machine-learning/voronoi-ward-prb-2017.ipynb) uses caching with MultiFeaturizer.